component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -299,58 +300,111 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
     public boolean sendRESTCall(String smsUrl, String httpMethod, String headerString, String payload,
                                 String httpResponse) throws IOException, AuthenticationFailedException {
 
-        HttpsURLConnection connection = null;
+        HttpURLConnection httpConnection = null;
+        HttpsURLConnection httpsConnection = null;
         String smsMessage = SMSOTPConstants.SMS_MESSAGE;
         try {
             smsUrl = smsUrl.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
                     smsMessage.replaceAll("\\s", "+") + otpToken);
             URL smsProviderUrl = new URL(smsUrl);
-            connection = (HttpsURLConnection) smsProviderUrl.openConnection();
-            connection.setDoInput(true);
-            connection.setDoOutput(true);
-            String[] headerList;
-            if (!headerString.isEmpty()) {
-                headerString = headerString.trim().replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
-                        smsMessage + otpToken);
-                headerList = headerString.split(",");
-                for (String aHeaderList : headerList) {
-                    String[] header = aHeaderList.split(":");
-                    connection.setRequestProperty(header[0], header[1]);
-                }
-            }
-            if (httpMethod.toUpperCase().equals(SMSOTPConstants.GET_METHOD)) {
-                connection.setRequestMethod(SMSOTPConstants.GET_METHOD);
-            } else if (httpMethod.toUpperCase().equals(SMSOTPConstants.POST_METHOD)) {
-                connection.setRequestMethod(SMSOTPConstants.POST_METHOD);
-                if (!payload.isEmpty()) {
-                    payload = payload.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg", smsMessage + otpToken);
+            String subUrl = smsProviderUrl.getProtocol();
+            if (subUrl.equals("https")) {
+                httpsConnection = (HttpsURLConnection) smsProviderUrl.openConnection();
+                httpsConnection.setDoInput(true);
+                httpsConnection.setDoOutput(true);
+                String[] headerList;
+                if (!headerString.isEmpty()) {
+                    headerString = headerString.trim().replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
+                            smsMessage + otpToken);
+                    headerList = headerString.split(",");
```

Comment:
No for all the apis endpoint call we need to specify the mobile number and message